### PR TITLE
adds metrics to repair QUIC endpoint

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
monitor repair QUIC endpoint

#### Summary of Changes
added metrics.

### CODE REVIEW

File: core/src/repair/quic_endpoint.rs [hunk 1]
```diff
@@ -471,19 +500,27 @@ async fn send_requests_task(
...
-async fn send_request_task(endpoint: Endpoint, connection: Connection, request: LocalRequest) {
+async fn send_request_task(
+    endpoint: Endpoint,
+    connection: Connection,
+    request: LocalRequest,
+    stats: Arc<RepairQuicStats>,
+) {
     if let Err(err) = send_request(endpoint, connection, request).await {
-        error!("send_request: {err:?}")
+        debug!("send_request: {err:?}");
```
Comment (lijunwangs) [inline core/src/repair/quic_endpoint.rs:522]: Can we output the destination address? That will be useful when examining the logs.

File: core/src/repair/quic_endpoint.rs [hunk 2]
```diff
@@ -698,6 +740,249 @@ impl<T> From<crossbeam_channel::SendError<T>> for Error {
...
+    router_try_send_error_full: AtomicU64,
+    write_error_connection_lost: AtomicU64,
+    write_error_stopped: AtomicU64,
+    write_error_unknown_stream: AtomicU64,
+    write_error_zero_rtt_rejected: AtomicU64,
+}
+
+async fn report_metrics_task(name: &'static str, stats: Arc<RepairQuicStats>) {
+    loop {
+        tokio::time::sleep(Duration::from_secs(2)).await;
```
Comment (lijunwangs) [inline core/src/repair/quic_endpoint.rs:773]: Can we define the "2" as a constant on the top?

File: core/src/repair/quic_endpoint.rs (hunk 1 above)
Comment (behzadnouri) [reply core/src/repair/quic_endpoint.rs]: done.

File: core/src/repair/quic_endpoint.rs (hunk 2 above)
Comment (behzadnouri) [reply core/src/repair/quic_endpoint.rs]: done.

**Review by lijunwangs — COMMENTED** [review]
lgtm

**Review by lijunwangs — APPROVED** [review]

**Pull request merged**